src/vs/workbench/contrib/preferences/browser/settingsTreeModels.ts

```diff
@@ -380,6 +382,11 @@ export class SettingsTreeSettingElement extends SettingsTreeElement {
 			return true;
 		}
 
+		if (!this.languageService.isRegisteredLanguageId(languageFilter)) {
```

Comment:
This is fine, but it seems like it would make more sense to check this upstream, just once for the language, instead of checking it for each of 1000 settings